src/sampling/fibonacci.jl

```diff
@@ -0,0 +1,61 @@
+"""
+	FibonacciSampling(n)
+
+Generate samples on a sphere using the Fibonacci lattice method.
+The parameter `n` specifies the number of points to generate.
+In the regular Fibonacci lattice method, the number ϕ is the golden ratio ((1 + √5)/2),
+but different numbers can be used, preferably irrational.
+
+## Example
+
+Sample a sphere with 1000 points:
+
+```julia
+sample(Sphere((0,0,0), 1), FibonacciSampling(100))
+
+# sample using π instead of the golden ration
+sample(Box((0,0),(1,1)), FibonacciSampling(100,π))
+"""
+struct FibonacciSampling{T<:Real} <: ContinuousSamplingMethod
+  n::Int
+  ϕ::T
+
+  function FibonacciSampling(n::Int, ϕ::T) where {T<:Real}
+    if n ≤ 0
+      throw(ArgumentError("Size must be positive"))
+    end
+    new{T}(n, ϕ)
+  end
+end
+
+# Outer constructor with default value for ϕ
+FibonacciSampling(n::Int) = FibonacciSampling(n, (1 + √5) / 2)
+
+function sample(::AbstractRNG, geom::Geometry, method::FibonacciSampling)
+  Iterators.map(0:(method.n - 1)) do i
+    # sample points in square
+    x, y = mod(i / method.ϕ, 1), i / method.n
+
+    # transform points to the geometry
+    Point(_fibonaccipointstogeom(geom, x, y)) |> _fibonaccigeomtransforms(geom)
```

Comment:
I believe that we can still simplify this algorithm further by relying on the already existing parametrization of geometries. Any geometry with paramdim(geom) == 2 can in theory support Fibonacci sampling. We just need to map the raw uv coords obtained with the expression

```julia
u, v = mod(i / method.ϕ, 1), i / method.n
```

to the uv coords of the geometry in question. For example, I believe that we can introduce a simple function `_distortion(g) = (u, v) -> (u, v)` that doesn't do anything by default, but can be specialized for Disk as in `_distortion(d::Disk) = (u, v) -> (u, sqrt(v))` (please double check the order and math.

If we do this and simply forward the distorted uv coords to the geometry, we won't need to re-implement the scale, translation, ... transforms. Makes sense?

Reply:
Pseudo-algorithm could look like:

```julia
function sample(geom::Geometry, method::FibonacciSampling)
  if paramdim(geom) != 2
    # throw error
  end

  f = _distortion(geom)

  function point(i)
    u, v = mod(i / method.ϕ, 1), i / method.n
    geom(f(u, v)...)
  end

  (point(i) for i in 1:method.n)
end